samples/bootloader/src/main.c

```diff
@@ -122,7 +122,13 @@ static void boot_from(const struct fw_info *fw_info)
 	__set_MSP(vector_table[0]);
 	__set_PSP(0);
 
-	/* Call reset handler. */
+	/* Ensure that data is written to the Register before the jump*/
+	__DSB();
```

Comment:
Since this is write to a register do you really need a dsb?
Also is isb really needed? There seems to be no concurrency here, won't it be enough to put a compiler barrier?

Reply:
You don't strictly need a DSB but it's very architecture dependent(according to ARM) so on a M7 with caches and multi-stage pipelines or http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0363e/Cfaccdhi.html dual issue processors one could theoretically be able to write to the register before the data being fetched`vector_table[0]` is ready. I strictly don't think it's needed but better safe than sorry.

Same goes for the ISB due to dual issue one could theoretically execute two instructions at once. So this change is just to ensure correct behavior on current and future products.

If you think it's overkill I can just remove it but I don't think it'll have a huge performance impact.

Reply:
Due to this you'll see that the 
```
// Barrier 
__DSB();
__ISB();
/* set the stack pointer */
_set_MSP(*external_value);
/* Change the vector table */
SCB->VTOR = ((uint32_t)new_vtor;
// Barrier
__DSB();
__ISB();
```
Pattern is quite common for ARM bootloaders.

Reply:
@sigvartmh 
I was wondering about the comment explaining usage of the barrier (`Ensure that data is written to the Register before the jump`). Registers do not need maintenance I think but it is true that DSB will ensure data is read from memory and moved into the register before the jump. If VTOR is something that needs to be stored maintenance is also needed.

Maybe changing a comment a bit would make things clear:
`Barriers to ensure all memory operations are done before the jump.` or something similar to focus on memory.

I don't think this is an overkill as I believe it is good to leave things clean before the jump to unknown.

Reply:
I do agree that my comments in that file are bit inaccurate I'll fix them up in this commit.

Reply:
ISBs are required here, but not DSBs, at least for Cortex-M.

Reply:
Removed DSB.